nilearn/glm/tests/test_first_level.py

```diff
@@ -573,6 +575,197 @@ def test_first_level_glm_computation_with_memory_caching():
+                                         tasks=['main'],
+                                         n_runs=[1])
+
+    RepetitionTime = 6.0
+    StartTime = 2.0
+    add_metadata_to_bids_dataset(
+        bids_path=tmp_path / bids_path,
+        metadata={"RepetitionTime": RepetitionTime,
+                  "StartTime": StartTime})
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        models, *_ = first_level_from_bids(
+            dataset_path=str(tmp_path / bids_path),
+            task_label='main',
+            space_label='MNI',
+            img_filters=[('desc', 'preproc')],
+            slice_time_ref=None,)
+        assert models[0].t_r == RepetitionTime
+        assert models[0].slice_time_ref == StartTime / RepetitionTime
+
+
+def test_first_level_from_bids_get_RepetitionTime_from_derivatives(tmp_path):
+    """Only RepetitionTime is provided in derivatives.
+
+    Warning about missing StarTime time in derivatives.
+    slice_time_ref cannot be inferred: defaults to 0.
+    """
+    bids_path = create_fake_bids_dataset(base_dir=tmp_path,
+                                         n_sub=10,
+                                         n_ses=1,
+                                         tasks=['main'],
+                                         n_runs=[1])
+    RepetitionTime = 6.0
+    add_metadata_to_bids_dataset(
+        bids_path=tmp_path / bids_path,
+        metadata={"RepetitionTime": RepetitionTime})
+
+    with pytest.warns(UserWarning,
+                      match="StartTime' not found in file"):
+        models, *_ = first_level_from_bids(
+            dataset_path=str(tmp_path / bids_path),
+            task_label='main',
+            space_label='MNI',
+            slice_time_ref=None,
+            img_filters=[('desc', 'preproc')])
+        assert models[0].t_r == 6.0
+        assert models[0].slice_time_ref == 0.
+
+
+def test_first_level_from_bids_get_StartTime_from_derivatives(tmp_path):
+    """Only StartTime is provided in derivatives.
+
+    Warning about missing repetition time in derivatives,
+    but RepetitionTime is still read from raw dataset.
+    """
+    bids_path = create_fake_bids_dataset(base_dir=tmp_path,
+                                         n_sub=10,
+                                         n_ses=1,
+                                         tasks=['main'],
+                                         n_runs=[1])
+    StartTime = 1.0
+    add_metadata_to_bids_dataset(
+        bids_path=tmp_path / bids_path,
+        metadata={"StartTime": StartTime})
+
+    with pytest.warns(UserWarning,
+                      match="RepetitionTime' not found in file"):
+        models, *_ = first_level_from_bids(
+            dataset_path=str(tmp_path / bids_path),
+            task_label='main',
+            space_label='MNI',
+            img_filters=[('desc', 'preproc')],
+            slice_time_ref=None)
+        assert models[0].t_r == 1.5
```

Comment:
Sorry, but where does this 1.5 come from ?

Reply:
ah yeah that definitely feels like a magic number: this is because the function that generate a BIDS dataset used for testing creates timeseries with a TR of 1.5 seconds: will explain than in the test docstring

Reply:
added a comment